### PR TITLE
(WIP) Code for source separation downstream (please don't merge)

### PR DESCRIPTION
**Pull request by HuangZiliAndy**
Hi Leo and everyone, I am trying to add the downstream source separation script for the LibriMix dataset. I will update about (1) the results I got and (2) the usage of this script later on (Maybe on Slack). Before this, I wonder if I should talk to some students from your team? I am not so sure about the usual way you check a downstream task. Thanks a lot! Also @sw005320 for his information.

### CODE REVIEW

**Comment by leo19941227** [comment]
Hi Zili!

Thanks for the PR!
Usually I run the training and testing myself to see if it is working.
So it will be nice if you help me update the usage of this downstream task in our internal README.
You can find that hackmd link from the `pinned message` in `implementation` section of our Slack :)

**Comment by leo19941227** [comment]
Hey Zili!

Thanks for the great contribution. I notice that we have to manually align the shift size for different upstream.
I am thinking that if this can be avoided by considering the `upstream_rate` (which is the down-sample rate on 16khz waveform.) here

https://github.com/s3prl/s3prl/blob/3a46a7a95ac29da18906f07f903bc740a530a4f9/downstream/runner.py#L85-L90

What do you think :)

**Comment by leo19941227** [comment]
Also, I am sorry that you did not know that during your implementation, our major development is located on the `develop` branch. Though `develop` has been merged back to `master` now, the interface for implementing downstream tasks is slightly different from the version you followed. However there are just some minor changes, and I think the current code just need some simple adjustment. You can have a look on the current interface here: [downstream/example/expert.py](https://github.com/s3prl/s3prl/blob/master/downstream/example/expert.py).

So the best way to do this might be that you pull the latest `master` branch and rebase (or merge) your separation branch to apply the latest interface. Then do some adjustment to make separation downstream runnable.

Really sorry about this :(
Hope this won't cause too much work for you. Thanks!

**Comment by HuangZiliAndy** [comment]
Hi Leo, the code is already changed to match the latest version. Thanks!

**Review by leo19941227 — COMMENTED** [review]
Thanks @HuangZiliAndy ! I only have a minor comment for now.
I am now running the experiment, and will merge it if the training works.
Many thanks!

File: downstream/separation_stft/configs/cfg.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    n_fft: 512
+    win_length: 512
+    window: "hann"
+    center: True
+
+  loaderrc:
+    num_workers: 4
+    train_batchsize: 8
+    eval_batchsize: 1
+    train_dir: /export/c12/hzili1/tools/s3prl_separation/downstream/separation_stft/data/wav16k/min/train-100
```
[the remark this answers is not in the document]
Comment (leo19941227) [reply downstream/separation_stft/configs/cfg.yaml:32]: Can you also update the `requirement.txt` at the root of this repo to add `asteroid==0.4.4`?

[the remark this answers is not in the document]
Comment (HuangZiliAndy) [reply downstream/separation_stft/configs/cfg.yaml]: Hi Leo, already fix that. Thanks!

**Pull request merged**